src/main.cpp

```diff
@@ -5205,10 +5586,41 @@ bool SendMessages(CNode* pto, bool fSendTrickle)
             NodeId staller = -1;
             FindNextBlocksToDownload(pto->GetId(), MAX_BLOCKS_IN_TRANSIT_PER_PEER - state.nBlocksInFlight, vToDownload, staller);
             BOOST_FOREACH(CBlockIndex *pindex, vToDownload) {
-                vGetData.push_back(CInv(MSG_BLOCK, pindex->GetBlockHash()));
-                MarkBlockAsInFlight(pto->GetId(), pindex->GetBlockHash(), consensusParams, pindex);
-                LogPrint("net", "Requesting block %s (%d) peer=%d\n", pindex->GetBlockHash().ToString(),
-                    pindex->nHeight, pto->id);
+                if (IsThinBlocksEnabled() && IsChainNearlySyncd()) {
```

Comment:
For examine, IsChainNearlySyncd() doesn't need to be defined in unlimited.cpp as when merging with other Bitcore products, it becomes an advert for unlimited effectively unless it's moved into a non-branded file, such as main.cpp.